src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -497,7 +497,8 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 			// {{SQL CARBON EDIT}}
 			return convertEditorInput(this.untitledEditorService.createOrGet(
 				untitledInput.filePath ? URI.file(untitledInput.filePath) : untitledInput.resource,
-				'sql',
+				(untitledInput.resource &&
```

Comment:
2 things:
- Put this logic on its own line, not inside this calling code
- Let's lookup if the existing resource is in the notebookService (aka are we tracking this) rather than guessing based on if we happened to call it "untitled" or "notebook". Those aren't scalable.

Reply:
Done